src/sentry/static/sentry/app/components/emptyStateWarning.tsx

```diff
@@ -2,70 +2,58 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import styled from '@emotion/styled';
 
-import HeroIcon from 'app/components/heroIcon';
 import EmptyMessage from 'app/views/settings/components/emptyMessage';
 import {IconWarning} from 'app/icons';
+import SvgIcon from 'app/icons/svgIcon';
 import space from 'app/styles/space';
 
 type Props = {
   small?: boolean;
   children?: React.ReactNode;
-  withIcon?: boolean;
+  icon?: boolean | typeof SvgIcon;
   className?: string;
 };
 
-const EmptyStateWarning = ({
-  small = false,
-  withIcon = true,
-  children,
-  className,
-}: Props) =>
+/**
+ * This component differs from the EmptyMessage in that it specializes for empty
+ * states where something is problematic.
+ *
+ * e.g. A users search has resulted in no entries, nothing matches a users
+ * filter
+ *
+ * This should NOT be used as an empty state for no items, or when an inbox is
+ * empty (a good thing).
+ */
+const EmptyStateWarning = ({small = false, icon = true, children, className}: Props) =>
   small ? (
     <EmptyMessage className={className}>
       <SmallMessage>
-        {withIcon && <StyledIconWarning color="gray500" size="lg" />}
+        {icon ? <IconWarning color="gray500" size="lg" /> : icon}
         {children}
       </SmallMessage>
     </EmptyMessage>
   ) : (
-    <EmptyStreamWrapper data-test-id="empty-state" className={className}>
-      {withIcon && <HeroIcon src="icon-circle-exclamation" size="54" />}
-      {children}
-    </EmptyStreamWrapper>
+    <EmptyMessage
+      icon={icon ? <IconWarning size="48" color="gray400" /> : icon}
```

Comment:
I find this confusing 🧐 maybe it would be better to make icon accept only SVGIcon `icon?:  typeof SvgIcon;` and here write something like` icon || <IconWarning size="48" color="gray400" /> ` ?

Reply:
I agree, this is confusing (possibly not correct?), the only possible values for `icon` are  `<IconWarning>` or a falsey value -- it'll never pass the SVGIcon

Reply:
I think the type is wrong right now, anyway the real question is, should we even allow icon customization in this component?

Reply:
Previously we didn't fwiw

Reply:
If we don't need it now, lets not add it. We can always revise this component in the future when we need to change it.

Reply:
👍 I'll remove. I think I added this because I was going to use it, but decided against it.